### PR TITLE
DataViews: Type the ItemActions component

### PR DESCRIPTION
**Pull request by youknowriad**
Related #55083 
Follow-up to #61185

## What?

The DataViews package is a types heavy package. There's a lot of structures that need to be documented properly. So this continues the effort to add explicit types to the package. This PR focuses on typing the ItemsActions component which forces us to define our Action types properly.

## Testing instructions

There's no functional change, it's essentially a code quality + documentation change.

### CODE REVIEW

**Comment by sirreal** [comment]
I see some CI failures on this PR - I _think_ they might be fixed by upgrading our react types (https://github.com/WordPress/gutenberg/pull/60796). It seems like ariakit expects a different version of react types than we have.

**Review by sirreal — APPROVED** [review]
This looks good, I suspect the CI failures are unrelated.

File: packages/dataviews/src/item-actions.tsx [hunk 1]
```diff
@@ -36,18 +76,25 @@ function ButtonTrigger( { action, onClick } ) {
...
-function DropdownMenuItemTrigger( { action, onClick } ) {
+function DropdownMenuItemTrigger( {
+	action,
+	onClick,
+}: DropdownMenuItemTriggerProps ) {
 	return (
 		<DropdownMenuItem
 			onClick={ onClick }
-			hideOnClick={ ! action.RenderModal }
+			hideOnClick={ 'RenderModal' in action }
```
Comment (oandregal) [inline packages/dataviews/src/item-actions.tsx]: Isn't this inverting the existing logic?

File: packages/dataviews/src/item-actions.tsx [hunk 2]
```diff
@@ -78,11 +125,14 @@ function ActionWithModal( { action, item, ActionTrigger } ) {
...
-function ActionsDropdownMenuGroup( { actions, item } ) {
+function ActionsDropdownMenuGroup( {
+	actions,
+	item,
+}: ActionsDropdownMenuGroupProps ) {
 	return (
 		<DropdownMenuGroup>
 			{ actions.map( ( action ) => {
-				if ( !! action.RenderModal ) {
+				if ( 'RenderModal' in action ) {
```
Comment (oandregal) [inline packages/dataviews/src/item-actions.tsx:166]: This is more readable, I like it. Is this related to any TypeScript thing I should be aware of or just an unrelated change that was nice to have?

File: packages/dataviews/src/item-actions.tsx (hunk 1 above)
Comment (youknowriad) [reply packages/dataviews/src/item-actions.tsx]: Good catch :)

File: packages/dataviews/src/item-actions.tsx (hunk 2 above)
Comment (youknowriad) [reply packages/dataviews/src/item-actions.tsx:166]: yes, basically when you do this, typescript is able to figure out the type of action that has the "RenderModal" defined. If you keep the old check, typescript will complain that RenderModal may or may not be defined.

**Comment by sirreal** [comment]
I'm squash-merging #61486 here to confirm that fixes all the issues we're seeing here. It's not intended to be landed as part of this PR.

EDIT: I should've set this to draft before it pinged all the codeowners. Sorry for the noise, folks!

File: packages/dataviews/src/types.ts [hunk 3]
```diff
@@ -164,3 +164,78 @@ export interface ViewList extends ViewBase {
...
+	 * The label of the action.
+	 */
+	label: string;
+
+	/**
+	 * The icon of the action. (Either a string or an SVG element)
+	 * This should be IconType from the components package
+	 * but that import is breaking typescript build for the moment.
+	 */
+	icon?: any;
```
Comment (youknowriad) [inline packages/dataviews/src/types.ts:184]: So it turns out that it was the import of `IconType` from the components package that was breaking the build in this PR. Using any temporarily should fix it.

**Comment by youknowriad** [comment]
The failing linting test here is already failing on trunk (order CSS property). It should be addressed separately.

File: packages/dataviews/src/types.ts (hunk 3 above)
Comment (sirreal) [reply packages/dataviews/src/types.ts:184]: 😆 Just when #61486 is almost ready.

Comment (sirreal) [reply packages/dataviews/src/types.ts:184]: Some detail:

I think this works around the issue that #61486 addresses. The problem was a symptom of the `skipLibCheck` _not_ being present in this package's tsconfig. Because of problems in library types, anything that depended on the components package was effectively "infected" by invalid library types and would then require `skipLibCheck`.
